nvtabular/workflow/workflow.py

```diff
@@ -148,6 +149,26 @@ def fit(self, dataset: Dataset):
                     ddf=_transform_ddf(ddf, workflow_node.parents)
                 )
 
+                # Check for additional input columns that aren't generated by parents
+                # and fetch them from the root dataframe
+                if workflow_node.parents:
+                    parent_output_cols = sum(
+                        [parent.output_columns for parent in workflow_node.parents],
+                        nvtabular.ColumnSelector(),
+                    )
+                    addl_input_cols = set(workflow_node.input_columns.names) - set(
+                        parent_output_cols.names
+                    )
+
+                    if addl_input_cols:
+                        # transformed_ddf = _concat_columns(
+                        #   [transformed_ddf, ddf[list(addl_input_cols)]]
+                        # )
+                        transformed_ddf = dask.dataframe.multi.concat(
```

Comment:
I do think this `multi.concat` call is a bit dangerous.  You are effectively assuming that you can concatenate a list of columns from the original ddf onto a transformed version of that ddf.  Given that we do not track `divisions` in NVTabular, I am surprised that this command works without setting `ignore_unknown_divisions=True`.  However, even if the command does "work", we are assuming that the number of rows cannot change within any NVTabular transformation.  I suspect this is not a promise we can/should make.

I think the only reliable solution would be to select the columns before `_transform_ddf` is called.  Is there an efficient way that we can construct the union of `addl_input_cols` and the transformation columns before `_transform_ddf` is called?

Reply:
if we do the concatenation before the `_transform_ddf` and lets say the transform is going to add rows, would that also break because we would still use selector to pull out target columns and then only transform those columns. This would cause problems on the reassembly, wouldn't it? Or are we now presuming that the ddf that goes into transform_ddf will always work on the entire input ddf?

Reply:
I'm not immediately sure if there's an efficient way to compute the columns ahead, but we'll see what we can do to refactor in that direction

Reply:
>if we do the concatenation before the _transform_ddf and lets say the transform is going to add rows, would that also break because we would still use selector to pull out target columns and then only transform those columns.

Right. The transformations would not be allowed to change the row-count of individual columns.  They can depend on individual columns, but would need to transform the DataFrame as a whole.  However, this has always been an implicit requirement.